.github/workflows/integration-test-workflow.yml

```diff
@@ -0,0 +1,61 @@
+name: "Integration test workflow"
+
+on:
+  workflow_call:
+    inputs:
+      main-branch-name:
+        required: false
+        type: string
+        default: main
+      runs-on:
+        required: false
+        type: string
+        default: ubuntu-latest
+      working-directory:
+        required: false
+        type: string
+
+jobs:
+  main:
+    runs-on: ${{ inputs.runs-on }}
+    name: Run
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout [Pull Request]
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          # By default, PRs will be checked-out based on the Merge Commit, but we want the actual branch HEAD.
+          ref: ${{ github.event.pull_request.head.sha }}
+          # We need to fetch all branches and commits so that Nx affected has a base to compare against.
+          fetch-depth: 0
+
+      - uses: actions/checkout@v2
+        name: Checkout [Default Branch]
+        if: ${{ github.event_name != 'pull_request' }}
+        with:
+          # We need to fetch all branches and commits so that Nx affected has a base to compare against.
+          fetch-depth: 0
+
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v2
```

Comment:
Can we reference the action locally instead of by version number here?

Reply:
Oh, yes, we should